src/test/java/plugins/FavoriteTest.java

```diff
@@ -0,0 +1,43 @@
+package plugins;
+
+import org.jenkinsci.test.acceptance.Matchers;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.plugins.mock_security_realm.MockSecurityRealm;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.GlobalSecurityConfig;
+import org.jenkinsci.test.acceptance.po.User;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+
+@WithPlugins({"favorite", "mock-security-realm"})
+public class FavoriteTest extends AbstractJUnitTest {
+    public static final String USER = "dev";
+
+    @Test
+    public void shouldBeAbleToFavoriteAJob() throws Exception {
+        jenkins.jobs.create(FreeStyleJob.class, "my-project");
+
+        final GlobalSecurityConfig security = new GlobalSecurityConfig(jenkins);
+        security.open();
+        MockSecurityRealm realm = security.useRealm(MockSecurityRealm.class);
+        realm.configure(USER);
+        security.save();
+
+        jenkins.login().doLogin(USER);
+
+        waitFor(by.id("fav_my-project")).click();
+        waitFor(by.css(".icon-fav-active"));
+
+        final User user = new User(jenkins, USER);
+        jenkins.visit(user.getConfigUrl().toString());
+        waitFor(by.id("favorites"));
+        waitFor(by.id("fav_my-project")).click();
```

Comment:
should we not have a check that the favoutite is actually checked so that the page here reflects the Jenkins job view.

Reply:
The line 36 actually does that. If the span with `fav_my-project` is not present on the page, it means that the project is not in the user favorite.